Lib/re/_parser.py

```diff
@@ -49,7 +49,8 @@
     r"\S": (IN, [(CATEGORY, CATEGORY_NOT_SPACE)]),
     r"\w": (IN, [(CATEGORY, CATEGORY_WORD)]),
     r"\W": (IN, [(CATEGORY, CATEGORY_NOT_WORD)]),
-    r"\Z": (AT, AT_END_STRING), # end of string
+    r"\z": (AT, AT_END_STRING), # end of string
+    r"\Z": (AT, AT_END_STRING), # end of string (deprecated)
```

Comment:
Should we use "deprecated" language here if we're not deprecating `\Z`?

Reply:
What words do you suggest? "legacy"? "obsolete"?

Due to its subtle difference from other common implementations, it is very likely that `\Z` will be deprecated and removed in future, when it will be safe. It is kept for compatibility with older Python versions.